opencensus-shim/src/integrationTest/java/io/opentelemetry/opencensusshim/IntegrationTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opencensusshim;
+
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.Tracing;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import org.junit.jupiter.api.Test;
+
+// todo remove if/once the underlying issue is agreed upon
```

Comment:
is this `todo` intended to be merged? If so, then we need to have a GH issue created to fix it.

Reply:
Thank you for the reminder; no, it is not intended to be merged. With https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7488 in play, this class is solved elsewhere and I can remove it. I'm going to spend a little time adding to any existing tests in both PRs to ensure better overall coverage of cases I've either uncovered or can conceive.

Reply:
There is however, [this todo](https://github.com/open-telemetry/opentelemetry-java/pull/4900/files#diff-89535d3bd4ff4b4ea1d657e394546f3c1127af7eea4644504008729ae520d38fR36), which exists as a result of comments I made in the [original issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6876). Summarized as a question: "can and should the javaagent be changed to not require the `AgentSpan` type internally and thereby relieve shim/others implementations of needing to proxy calls to the specific javaagent-minted `Span`s?"

Not a small body of work, to be sure. But if the policy is one-issue-per-todo, wanted to either a) ensure adequate coverage or b) decide it's not worth tracking and remove it. WDYT?

Reply:
Since this test doesn't seem to have any assertions in it, could we beef it up to assert what is expected, at least? I'm really not familiar with this issue much at all... I just want to make sure we don't leave un-tracked work left to be done.

Reply:
Perfectly understandable. This test was just a stand-in to verify some assumptions before we introduced the [dedicated integration tests PR](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7488). I've now removed it in favor of a unit test which ensures the methods are all appropriately proxied. I've also uncovered some outliers which weren't caught by the original shim tests and will be adding those as well.